tests/ValidateSbrpTests/ValidateSbrpTests.cs

```diff
@@ -0,0 +1,170 @@
+            }
+            finally
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+    }
+
+    [Fact]
+    public async Task CheckForSignature()
+    {
+        var trustProviders = new List<ISignatureVerificationProvider>()
+        {
+            new SignatureTrustAndValidityVerificationProvider()
+        };
+        var verifier = new PackageSignatureVerifier(trustProviders);
+
+        var settings = SignedPackageVerifierSettings.GetDefault();
+
+        foreach (var package in Packages)
+        {
+            bool isSigned = await IsPackageSigned(package, verifier, settings);
+            Assert.False(isSigned, $"{package} is signed. Signed packages are not allowed in source-build-reference-packages.");
+        }
+    }
+
+    private bool HasSbrpAttribute(MetadataReader reader)
+    {
+        return reader.CustomAttributes.Select(attrHandle => reader.GetCustomAttribute(attrHandle)).Any(attr => IsAttributeSbrp(reader, attr));
+    }
+
+    private bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
+    {
+        string attributeType = string.Empty;
+
+        if (attr.Constructor.Kind == HandleKind.MemberReference)
+        {
+            var mref = reader.GetMemberReference((MemberReferenceHandle)attr.Constructor);
+            if (mref.Parent.Kind == HandleKind.TypeReference)
+            {
+                var tref = reader.GetTypeReference((TypeReferenceHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tref.Namespace)}.{reader.GetString(tref.Name)}";
+            }
+            else if (mref.Parent.Kind == HandleKind.TypeDefinition)
+            {
+                var tdef = reader.GetTypeDefinition((TypeDefinitionHandle)mref.Parent);
+                attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+            }
+        }
+        else if (attr.Constructor.Kind == HandleKind.MethodDefinition)
+        {
+            var mdef = reader.GetMethodDefinition((MethodDefinitionHandle)attr.Constructor);
+            var tdef = reader.GetTypeDefinition(mdef.GetDeclaringType());
+            attributeType = $"{reader.GetString(tdef.Namespace)}.{reader.GetString(tdef.Name)}";
+        }
+
+        if (attributeType == SbrpAttributeType)
+        {
+            var decodedValue = attr.DecodeValue(DummyAttributeTypeProvider.Instance);
+            try
+            {
+                return decodedValue.FixedArguments[0].Value.ToString() == "source" && decodedValue.FixedArguments[1].Value.ToString() == "source-build-reference-packages";
+            }
+            catch
+            {
+                throw new InvalidOperationException($"{SbrpAttributeType} is not formatted properly with a key, value pair.");
+            }
+        }
+
+        return false;
+    }
+
+    public async Task<bool> IsPackageSigned(string packagePath, PackageSignatureVerifier verifier, SignedPackageVerifierSettings settings)
```

Comment:
Convention is for async methods to have an `Async` suffix:

```suggestion
    public async Task<bool> IsPackageSignedAsync(string packagePath, PackageSignatureVerifier verifier, SignedPackageVerifierSettings settings)
```

Reply:
Changed in https://github.com/dotnet/source-build-reference-packages/commit/ce1a70dc071c6a5adce2b2d0ca6285031ee12258